client/my-sites/site-settings/wrap-settings-form.jsx

```diff
@@ -95,6 +95,7 @@ const wrapSettingsForm = getFormSettings => SettingsForm => {
 		handleToggle = name => () => {
 			this.props.trackEvent( `Toggled ${ name }` );
 			this.props.updateFields( { [ name ]: ! this.props.fields[ name ] } );
+			this.props.markChanged();
```

Comment:
This overlaps with what we discussed in #10412 

We could, but this would a big change compared to the current behaviour of these forms. It means Relying on the `props` to compute the "original" values, so we consider that receiving a new prop is equivalent to "the original value changed". Thus, we can update the dirtyFields and mark the form changed or not.

This is wise but changes the behaviour of the form, since we could revert the "dirty" status of fields without saving, the form value could change if we receive new props after the dirtyness reset. (which is not the case for now).

I'm planning to try this approach as well in a future PR.

Btw, this particular change is already merged by #10690